crates/provider-wadm/src/lib.rs

```diff
@@ -0,0 +1,624 @@
+        let connection = get_connection();
+        serve(
+            &connection.get_wrpc_client(connection.provider_key()),
+            provider,
+            shutdown,
+        )
+        .await
+    }
+
+    /// Build a [`WadmProvider`] from [`HostData`]
+    pub fn from_host_data(host_data: &HostData) -> WadmProvider {
+        let config = WadmConfig::from_map(&host_data.config);
+        if let Ok(config) = config {
+            WadmProvider {
+                default_config: config,
+                ..Default::default()
+            }
+        } else {
+            warn!("Failed to build connection configuration, falling back to default");
+            WadmProvider::default()
+        }
+    }
+
+    /// Attempt to connect to nats url (with jwt credentials, if provided)
+    async fn connect(
+        &self,
+        cfg: WadmConfig,
+        component_id: &str,
+    ) -> anyhow::Result<NatsClientBundle> {
+        let mut opts = match (cfg.auth_jwt, cfg.auth_seed) {
+            (Some(jwt), Some(seed)) => {
+                let seed = KeyPair::from_seed(&seed).context("failed to parse seed key pair")?;
+                let seed = Arc::new(seed);
+                async_nats::ConnectOptions::with_jwt(jwt, move |nonce| {
+                    let seed = seed.clone();
+                    async move { seed.sign(&nonce).map_err(async_nats::AuthError::new) }
+                })
+            }
+            (None, None) => async_nats::ConnectOptions::default(),
+            _ => bail!("must provide both jwt and seed for jwt authentication"),
+        };
+        if let Some(tls_ca) = &cfg.tls_ca {
+            opts = add_tls_ca(tls_ca, opts)?;
+        } else if let Some(tls_ca_file) = &cfg.tls_ca_file {
+            let ca = fs::read_to_string(tls_ca_file)
+                .await
+                .context("failed to read TLS CA file")?;
+            opts = add_tls_ca(&ca, opts)?;
+        }
+
+        // Use the first visible cluster_uri
+        let url = cfg.cluster_uris.first().unwrap();
+
+        // Override inbox prefix if specified
+        if let Some(prefix) = cfg.custom_inbox_prefix {
+            opts = opts.custom_inbox_prefix(prefix);
+        }
+
+        let client = opts.name("Wadm Provider").connect(url).await?;
+
+        // Connections
+        let mut sub_handles = Vec::new();
+        let subscription_subject = format!(
+            "{}.{}.{}",
+            WADM_STATUS_API_PREFIX, cfg.lattice, cfg.app_name
+        );
+        sub_handles.push((
+            subscription_subject.clone(),
+            self.subscribe(&client, component_id, subscription_subject, None)
+                .await?,
+        ));
+        Ok(NatsClientBundle {
+            client,
+            sub_handles,
+        })
+    }
+
+    /// Add a regular or queue subscription
+    async fn subscribe(
+        &self,
+        client: &async_nats::Client,
+        component_id: &str,
+        sub: impl ToSubject,
+        queue: Option<String>,
```

Comment:
I don't think you're using this anymore? Guessing it is copy/paste left over from the NATS code